dev/TeachingTip/TeachingTip.cpp

```diff
@@ -601,6 +597,12 @@ bool TeachingTip::PositionUntargetedPopup()
     return tipDoesNotFit;
 }
 
+void TeachingTip::TargetEffectiveViewportChanged(winrt::FrameworkElement const&,winrt::EffectiveViewportChangedEventArgs const&) {
+    // Reposition checks wether the teachingtip is open,
+    // so just rely on RepositionPopup and don't waste time checking here
+    RepositionPopup();
```

Comment:
If we do decide to change this behavior there is more we would need to do around this event. For one the effective view port changed event is only available on rs5+ so we need to check for windows version before attaching to it. another, we should really only be attached to the event while we are open, while this code functions it isn't hard to detatch the event while we aren't open, which for normal TeachingTip usage will be the majority of the time.

Reply:
Oh I didn't think about that. Should this detach/attach behavior also apply to registering to the windows maximized event? Or is this a something that is negligible as changing maximized state usually happens rather rarely?

Reply:
The WindowSizeChanged event attaches and detaches depending if we are open or not, so a window maximized event handler probably should as well. But you are right that these event are rare and it probably doesn't matter much.

---
In reply to: [346526001](https://github.com/microsoft/microsoft-ui-xaml/pull/1609#discussion_r346526001) [](ancestors = 346526001)

Reply:
I am sorry if this is a dumb question, but what event exactly should we use? I cant seem to find an event that we could listen/register to. I was also unable to find a window maximized event 😅

Reply:
Not a dumb question, I was assuming that the event existed but I don't think it does.  It looks like WindowSizeChanged is supposed to be firing for window maximizes, We'll have to debug to see why it is not..

Reply:
I think I found the problem: For some reason, when the window gets maximized/restored, it looks like the `WindowSizeChanged` event gets raised before the target has calculated its new position. 

While debugging, I noticed that the values we need to reposition the TeachingTip seem to be outdated. The target reports its old position, so we update our TeachingTip to render at the old position, instead of being at the new position, thus we are not displaying the TeachingTip correctly.

Reply:
Oh nice, that would explain the behavior for sure. I'm not sure what the right approach to fix that is.  It is possible that waiting some number of render tics before responding to window size changed events would fix the issue but it sounds like there may be an underlying framework bug that should just be fixed instead.

Reply:
If there is an easy way to wait until everything has rendered and updated its position, I think this would be a good idea. If that is not the case I think, waiting a certain amount of time might be a problematic fix (what if rendering takes too long?). But I think this sounds more like a bug in the underlying framework.

Reply:
@jevansaks I'm not sure what the best way foreword is here. Open a framework bug and dupe this issue to it? Try and find a work around for teaching tip in the mean time?

Reply:
Sorry I didn't see this sooner. 

So on window maximized we sometimes get a WindowSizeChanged event where the CoreWindow.Size() is still the old size? I notice that the TeachingTip code is not using the event args from the SizeChanged event. What if you use that size instead of always going to CoreWindow.Size()? I could believe that the property is lagging behind, but the event arg info should be correct.

Reply:
We actually don't need the "CoreWindow.Size()" (at least for a targeted TeachingTip). When the window size changes, we ask the TeachingTip's target where it is inside the window. Unfortunately this happens before the target knows it new position, which ends in the TeachingTip rendering at the old position.

Reply:
Oh, I see. So in that case we've probably queued layout changes but those won't happen until the next "Tick". In order to calculate results after everything has settled it's probably easiest to just do the work after the Tick (after the CompositionTarget.Rendering event fires). We have a helper to call a lambda just once after the next Tick, it's called `SharedHelpers::QueueCallbackForCompositionRendering`. Try that?

Reply:
Yes, that sounds like a good idea. Will try that as soon as I have time.